anvil/src/winit.rs

```diff
@@ -129,24 +129,46 @@ pub fn run_winit() {
     #[cfg(feature = "debug")]
     let mut fps_element = FpsElement::new(fps_texture);
 
-    let data = {
-        #[cfg(feature = "egl")]
-        let dmabuf_state = if backend.renderer().bind_wl_display(&display.handle()).is_ok() {
-            info!("EGL hardware-acceleration enabled");
+    let render_node = EGLDevice::device_for_display(backend.renderer().egl_context().display())
+        .and_then(|device| device.try_get_render_node());
+
+    let dmabuf_state = match render_node {
+        Ok(Some(node)) => {
             let dmabuf_formats = backend.renderer().dmabuf_formats().cloned().collect::<Vec<_>>();
-            let mut state = DmabufState::new();
-            let global = state.create_global::<AnvilState<WinitData>>(&display.handle(), dmabuf_formats);
-            Some((state, global))
-        } else {
+            let dmabuf_default_feedback = DmabufFeedbackBuilder::new(node.dev_id(), dmabuf_formats)
+                .build()
+                .unwrap();
+            let mut dmabuf_state = DmabufState::new();
+            let dmabuf_global = dmabuf_state.create_global_with_default_feedback::<AnvilState<WinitData>>(
+                &display.handle(),
+                &dmabuf_default_feedback,
+            );
+            Some((dmabuf_state, dmabuf_global, dmabuf_default_feedback))
+        }
+        Ok(None) => {
+            warn!("failed to query render node, dmabuf will be unavailable");
```

Comment:
Couldn't we do v3 at least, if the `EGLDevice`-extensions are not available or if the renderer has no `render_node` (either because it is an old driver or because we are using llvmpipe for example?).

Reply:
I guess this is not super important, given we still enable `wl_drm` (at least), but we could add at least a comment here about it.

Reply:
Yeah, but I added the fallback path to initialize v3 and added a note that mesa requires either v4 or wl_drm